README.md

```diff
@@ -1,2 +1,85 @@
 # business-analytics
 Tools to produce Riff Business analytic reports on RiffEdu and RiffPlatform deployments
+
+This repository uses a third-party package called [Metabase][] to visualize usage metrics for Mattermost. Two docker containers are run; one using the metabase image and another using the
+official mysql image. Commands are provided to restore sql database backups from Mattermost to the mysql container.
+
+## Getting Started
+
+These instructions will get you a copy of the project up and running on your local machine for development and testing purposes.
+
+### Database Backups
+
+Download sql backups from the Riff Learning backup location (ask for the URL if you don't have it), and place all .sql files in the folder called 'database-backups' in the root of this repo.
+
+### Prerequisites
+
+- [docker CE][docker-install] (Community Edition)
+- [docker-compose][docker-compose-install]
+- git
+- make
+
+
+## Running the Software
+The first step is to clone this repository. The following commands are defined in a Makefile, and can be run as follows:
+
+```
+make up
+```
+This command will create the mysql and metabase containers and show the logs for each
+
+```
+make up-detach
+```
+This command will create the mysql and metabase containers in detached mode (no logs).
+
+```
+make down
+```
+This command will stop and remove the mysql and metabase containers. The mysql volume containing the sql backups will
+persist locally however, so only make up or make up-detach is needed to start the software again.
+
+#### Restoring Database Backups
+Once the `make up` command has completed (about 2-3 minutes), open a new terminal tab and run `bin/restore-database-backups.sh`. It will restore databases from all .sql files in the folder ./database-backups. You may need to run `chmod +x ./bin/restore-database-backups.sh`, to make the file executable.
```

Comment:
The issue is that just because the `make up` has successfully started all the containers doesn't mean that the mysql instance running in its container is ready for the restore to be started. Determining that is tricky.

Reply:
Yeah. There are solutions for detecting when the sql container is ready for a db backup to be restored. I guess if we determine that it's a priority for everything to happen with one command (which WOULD be cleaner), and if I get the go ahead, I can spend some time and implement that.

Reply:
That makes more sense to me. Maybe try adding: `&& sleep 5s && ./bin/restore-database-backups.sh` as a hacky workaround? I like the single command build even if it's not completely correct.

Reply:
you don't need that all in a single line, although putting it all in a single line means it all runs in the same subshell.
But in this case just run those all sequentially.
However, I really think this is a future optimization (you can figure out how to improve the base script to check that the container is started, then test if the container is ready, then do the restore)

Reply:
The swiftest way I found to do this was to run a looped grep on the mysql container logs to check for a line of text that signifies the container is ready. 

Ok so maybe I will hold off on this for now?